### PR TITLE
Arizona cash assistance earned income

### LINKED ISSUE #4320: Arizona Cash Assistance earned income

### PR DESCRIPTION
**Pull request by liwenxin491**
Fixes #4320

Co-authored-by: Pavel Makarchuk <PavelMakarchuk@users.noreply.github.com>

### CODE REVIEW

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/older.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: Arizona subtracts the money spent on care for each child(2 years old or older) up to this threshold
+metadata:
+  unit: currency-USD
+  label: Arizona older child care expense income subtraction threshold
+  period: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/older.yaml:5]:
```suggestion
  period: month
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/older.yaml:5]: all disregard amounts should be monthly

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        # Disabled adult expenses
+        disabled_adult_care = spm_unit("care_expenses", period)
+        # Determine the expense discount
+        p = parameters(period).gov.states.az.hhs.tanf.eligibility.income.earned
+        young_child = child_age < p.child_age
+        uncapped_expense_disregard = spm_unit.sum(where(young_child, p.younger, p.older))
+        capped_expense_disregard = min_(uncapped_expense_disregard, childcare)
+        # Determine the disabled adult care expense after discount
+        disabled_care_expense_discount = min_(disabled_adult_care, p.adult)
+        return (income - p.flat) * p.percentage - capped_expense_disregard - disabled_care_expense_discount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py:27]: break this formula out in multiple steps

File: policyengine_us/tests/policy/baseline/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      person1:
+        age: 30
+      person2: 
+        age: 12 
+      person3: 
+        age: 1
+    spm_units:
+      spm_unit:
+        members: [person1, person2, person3]
+
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.yaml:87]: finish this test

File: policyengine_us/tests/policy/baseline/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    state_code: AZ
+  output:
+    az_tanf_earned_income: 642
+
+- name: Spend 180 on an adult with disability
+  period: 2023
+  absolute_error_margin: 0.01
+  input:
+    earned_income: 15_000
+    childcare: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.yaml:66]: use the variable inputs from the variable file

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/adult.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+description: Arizona subtracts the money spent on care for an adult with a disability up to this threshold
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/adult.yaml:1]: similar language in all yaml files 
- arizona does xyz under the cash assistance program.

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/adult.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+metadata:
+  unit: currency-USD
+  label: Arizona disable adults care expense income subtraction threshold
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/adult.yaml:4]: mention cash assistance in all labels

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py [hunk 7]
```diff
@@ -10,18 +10,27 @@ class az_tanf_earned_income(Variable):
...
         # Determine the disabled adult care expense after discount
-        disabled_care_expense_discount = min_(disabled_adult_care, p.adult)
-        return (income - p.flat) * p.percentage - capped_expense_disregard - disabled_care_expense_discount
+        adult_discount_threshold=where(member_age > 18, p.adult , 0)
+        capped_adult_expense_disregard = min_(spm_unit.sum(monthly_care_expense), adult_discount_threshold)        
+
+        # Calculate capped disabled adult care expense
+        # use age eligibility variable later
+        eligible_child = person("az_tanf_eligible_child",period)
+        whether_young_child = member_age < p.child_age
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py:32]:
```suggestion
        young_eligible_child = member_age < p.child_age
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+description: Arizona decreases the care expense subtraction threshold at this age under its cash assistance program
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml:1]:
```suggestion
description: Arizona decreases the care expense subtraction threshold at this age under its cash assistance program.
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml:1]: same in all descriptions

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+description: Arizona have a preliminary subtraction on earned income in calculation under its cash assistance program
+metadata:
+  unit: currency-USD
+  label: Arizona preliminary earned income subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml:4]:
```suggestion
  label: Arizona flat earned income subtraction
```

File: policyengine_us/tests/policy/baseline/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.yaml:1]: add a couple tests for edge cases - going to 0 etc.

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+    def formula(spm_unit, period, parameters):
+        # Earned income of the spm unit
+        income = add(spm_unit, period, ["earned_income"])
+        # Determine the expense discount
+        p = parameters(period).gov.states.az.hhs.tanf.eligibility.income.earned
+        # Yearly subtracted income disregard
+        yearly_flat_discount = p.flat * MONTHS_IN_YEAR
+        # Income after subtracting constant value and certain percentage
+        after_subtracted_income = max_(income - yearly_flat_discount, 0)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py:19]:
```suggestion
        income_after_flat_disregard = max_(income - yearly_flat_discount, 0)
```

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        after_discounted_income = after_subtracted_income * (1 - p.percentage)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py:20]:
```suggestion
        income_percentage_disregard = after_subtracted_income * (1 - p.percentage)
```

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        # Calculate countable earned income by further subtracting earned income disregard
+        earned_income_disregard = spm_unit(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py:22]:
```suggestion
        earned_income_care_expense_disregard = spm_unit(
```

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_disregard.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
+from policyengine_us.model_api import *
+
+
+class az_hhs_tanf_earned_income_disregard(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_disregard.py:4]:
```suggestion
class az_hhs_tanf_earned_income_care_expense_disregard(Variable):
```

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_disregard.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_disregard.py:1]: rename file az_hhs_tanf_earned_income_care_expense_disregard.py - same for test file

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/older.yaml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+description: Arizona subtracts the money spent on care for each child(2 years old or older) up to this threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/older.yaml:1]:
```suggestion
description: Arizona disregards this amount of care expenses for older children, under the cash assistance program.
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+description: Arizona decreases the care expense subtraction threshold at this age under its cash assistance program.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml:1]:
```suggestion
description: Arizona decreases the care expense disregard amount for children at or over this age under its cash assistance program.
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/younger.yaml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+description: Arizona subtracts the money spent on care for each child(under 2 years old) up to this threshold under the cash assistance program.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/younger.yaml:1]:
```suggestion
description: Arizona disregards this amount of care expenses for younger children, under the cash assistance program.
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml [hunk 17]
```diff
@@ -0,0 +1,11 @@
+description: Arizona have a preliminary subtraction on earned income in calculation under its cash assistance program.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml:1]:
```suggestion
description: Arizona disregards this amount of earned income under the cash assistance program.
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+metadata:
+  unit: currency-USD
+  label: Arizona Cash Assistance flat earned income subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml:4]:
```suggestion
  label: Arizona Cash Assistance flat earned income disregard
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/percentage.yaml [hunk 19]
```diff
@@ -0,0 +1,11 @@
+description: Arizona have a preliminary percentage discount on earned income in calculation
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/percentage.yaml:1]:
```suggestion
description: Arizona disregards this percentage of earned income after the flat amount disregard, under the cash assistance program.
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/percentage.yaml [hunk 20]
```diff
@@ -0,0 +1,11 @@
+description: Arizona have a preliminary percentage discount on earned income in calculation 
+metadata:
+  unit: /1
+  label: Arizona Cash Assistance preliminary earned income percentage discount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/percentage.yaml:4]:
```suggestion
  label: Arizona Cash Assistance preliminary earned income percentage disregard
```

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_care_expense_disregard.py [hunk 21]
```diff
@@ -0,0 +1,42 @@
+from policyengine_us.model_api import *
+
+
+class az_hhs_tanf_earned_income_care_expense_disregard(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "Disregard of the earned income for the Arizona Cash Assistance"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_care_expense_disregard.py:7]:
```suggestion
    label = "Arizona Cash Assistance care expense earned income disregard" 
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/adult.yaml [hunk 22]
```diff
@@ -0,0 +1,11 @@
+description: Arizona subtracts this amount of care expenses for disabled adults from earned income under the cash assistance program.
+metadata:
+  unit: currency-USD
+  label: Arizona Cash Assistance disable adults care expense disregard
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/adult.yaml:4]:
```suggestion
  label: Arizona Cash Assistance disabled adults care expense disregard
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml [hunk 23]
```diff
@@ -0,0 +1,11 @@
+description: Arizona decreases the care expense disregard amount for children at or over this age under its cash assistance program.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml:1]:
```suggestion
description: Arizona reduces the care expense disregard for children this age or older under its cash assistance program.
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml:1]: can we consolidate these four into a single scale parameter?

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml [hunk 24]
```diff
@@ -0,0 +1,11 @@
+description: Arizona disregards this amount of earned income under the cash assistance program.
+metadata:
+  unit: currency-USD
+  label: Arizona Cash Assistance flat earned income disregard
+  period: year
+  reference:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml:6]: can you find a primary source?

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml [hunk 25]
```diff
@@ -0,0 +1,11 @@
+description: Arizona disregards this amount of earned income under the cash assistance program.
+metadata:
+  unit: currency-USD
+  label: Arizona Cash Assistance flat earned income disregard
+  period: year
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml:5]: right? same for percentage
```suggestion
  period: month
```

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py [hunk 26]
```diff
@@ -0,0 +1,30 @@
+from policyengine_us.model_api import *
+
+
+class az_tanf_earned_income(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "Earned income for the Arizona Cash Assistance"
+    definition_period = YEAR
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income.py:8]: @PavelMakarchuk let's make all such programs monthly from here on
```suggestion
    definition_period = MONTH
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_age.yaml:1]: @MaxGhenis I think we can consolidate the amounts for children but would need to leave the adult amount separately as it depends on a different condition and is defined separately

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_care_expense_disregard.py [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+        person = spm_unit.members
+        age = person("age", period)
+        # Get the childcare and disabled adult care expenses
+        care_expenses = spm_unit("childcare_expenses", period)
+        monthly_care_expenses = care_expenses / MONTHS_IN_YEAR
+        # Determine the total eligible disregard
+        # The eligibility reuquirements consider wither children or disabled adults
+        young_eligible_child = age < p.child_age
+        is_child = person("az_tanf_eligible_child", period)
+        disabled_adult = person("is_disabled", period)
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/az/hhs/tanf/az_hhs_tanf_earned_income_care_expense_disregard.py:25]: and we can then delete the extra files regarding child age and amount

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_amounts.yaml [hunk 28]
```diff
@@ -0,0 +1,20 @@
+description: Arizona disregards this amount of care expenses for older children, under the cash assistance program.
+metadata:
+  unit: currency-USD
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_amounts.yaml:3]:
```suggestion
```

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_amounts.yaml [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+  type: single_amount
+  label: Arizona Cash Assistance care expense income subtraction threshold for children under or over age threshold
+  period: month
+  threshold_unit: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/care_expenses/child_amounts.yaml:7]:
```suggestion
  threshold_unit: year
  amount_unit: currency-USD
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml (hunk 24 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/az/hhs/tanf/eligibility/income/earned/flat.yaml:6]: @liwenxin491 could you find the numbers here and add it as a reference:
https://dbmefaapolicy.azdes.gov/index.html#page/FAA1/The_Cash_Assistance_(CA)_Program.html

**Comment by PavelMakarchuk** [comment]
Will be addressed by #4469
